types/ember__object/events.d.ts

```diff
@@ -1,34 +1,28 @@
-import { ObserverMethod } from '@ember/object/-private/types';
+import { ObserverMethod, Callback } from '@ember/object/-private/types';
 
 /**
  * Add an event listener
  */
-export function addListener<Context, Target>(
-    obj: Context,
-    key: keyof Context,
-    target: Target,
-    method: ObserverMethod<Target, Context>,
+export function addListener(
+    obj: object,
+    eventName: string,
+    target: object | Callback | null,
+    method?: Callback | string,
     once?: boolean
 ): void;
-export function addListener<Context>(
-    obj: Context,
-    key: keyof Context,
-    method: ObserverMethod<Context, Context>
-): void;
+
 /**
  * Remove an event listener
+ *
+ * Arguments should match those passed to `addListener`.
```

Comment:
this JSDoc stuff shows up in tooltips as users hover over various symbols and navigate within intellisense autocompletes. I tend to find that it's a better experience to find complete information (as opposed to a reference to another function's information) in these tooltips. 

What are your thoughts around the way to provide the best developer experience?